packages/project-access/src/project/cap.ts

```diff
@@ -0,0 +1,109 @@
+import { join } from 'path';
+import { FileName } from '../constants';
+import type { CapCustomPaths, csn, Package } from '../types';
+import { fileExists, readJSON } from '../file';
+import { loadModuleFromProject } from './moduleLoader';
+
+/**
+ * Returns true if the project is either a CAP Node.js or a CAP Java project.
+ *
+ * @param projectRoot - the root path of the project
+ * @param [packageJson] - optional: the parsed package.json object
+ * @returns - true if the project is a CAP project; false otherwise
+ */
+export async function isCapProject(projectRoot: string, packageJson?: Package): Promise<boolean> {
+    if ((await isCapNodeJsProject(projectRoot, packageJson)) || (await isCapJavaProject(projectRoot))) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Returns true if the project is a CAP Node.js project.
+ *
+ * @param projectRoot - the root path of the project
+ * @param [packageJson] - the parsed package.json object
+ * @returns - true if the project is a CAP Node.js project
+ */
+export async function isCapNodeJsProject(projectRoot: string, packageJson?: Package): Promise<boolean> {
```

Comment:
What would be a nice interface for supporting mem-fs? I see two options but I am unsure which to recommend:
- adding it as a 3rd parameter
- making the 1st parameter of type `string | { root: string; fs: Editor }`

Reply:
Actually, I don't think the function is well designed (same for `isCapProject`). If you already have a `package.json` then you want to know if it is node with or without cap. If you don't have the package.json yet, then you want to know if it is CAP Java, Cap Node, or sth. else (the function will not tell you if it is a non-cap node project).
Please think about the exact usage of the functions and maybe split them e.g. 
- check for an already loaded package.json if CAP is enabled or not
- get the project type for a folder: cap-node, cap-java, ...

Reply:
Changed this with https://github.com/SAP/open-ux-tools/pull/685/commits/601fd71425e1fc4cc3dc3c6fac51bedb17ef83af. 

We have now one new type:
```
type CapProjectType = 'CAPJava' | 'CAPNodejs';
```

and 3 exported functions:

```
isCapNodeJsProject(packageJson: Package): boolean
isCapJavaProject(projectRoot: string): Promise<boolean>
getCapProjectType(projectRoot: string): Promise<CapProjectType | undefined>
```